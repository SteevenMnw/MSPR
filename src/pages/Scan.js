//import liraries
import React from "react";
import { View, Text, StyleSheet } from "react-native";

import QrCode from "../components/QrCode";

// create a component
const Scan = () => {
  return <QrCode />;
};

// define your styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#76a6ef",
  },
  text: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#fff",
  },
});

//make this component available to the app
export default Scan;
