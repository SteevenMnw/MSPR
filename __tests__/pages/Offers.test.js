import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { expect, it } from "@jest/globals";

import Scan from "../../src/pages/Scan";

describe("<Home />", () => {
    it("has 1 child", () => {
      const tree = renderer.create(<Scan />).toJSON();
      expect(tree.children.length).toBe(1);
    });
  
    // Snapshop de l'ensemble de la page Home
    it("renders correctly", () => {
      const tree = renderer.create(<Scan />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});