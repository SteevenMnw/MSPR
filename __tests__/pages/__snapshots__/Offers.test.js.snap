// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offers /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E8E8E8",
      "flex": 1,
    }
  }
  testID="test_view_container"
>
  <Text
    style={
      Object {
        "color": "#ba473c",
        "fontSize": 40,
        "fontWeight": "bold",
        "paddingBottom": 5,
        "paddingLeft": 15,
        "paddingTop": 5,
      }
    }
    testID="test_text_title"
  >
    Mes offres
  </Text>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;

exports[`<Offers /> renders default elements and styles: test Container View style 1`] = `
Object {
  "backgroundColor": "#E8E8E8",
  "flex": 1,
}
`;

exports[`<Offers /> renders default elements and styles: test Title Text style 1`] = `
Object {
  "color": "#ba473c",
  "fontSize": 40,
  "fontWeight": "bold",
  "paddingBottom": 5,
  "paddingLeft": 15,
  "paddingTop": 5,
}
`;
